src/node_buffer.cc

```diff
@@ -701,6 +931,53 @@ void IndexOfNumber(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void IsNull(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!HasInstance(args[0]))
+    return env->ThrowTypeError("first arg should be a Buffer");
+
+  CHECK(args[0]->IsObject());
+  ARGS_THIS(args[0].As<Object>())
+
+  uint32_t offset = args[1]->Uint32Value();
+  CHECK_LE(offset, obj_length);
+
+  char* ptr = obj_data + offset;
+
+  args.GetReturnValue().Set(ptr == nullptr);
```

Comment:
Are you saying basically just map JS `null` to, and remove the NULL buffer concept altogether (i.e. readPointer() would return JS `null` when a NULL pointer is read)? Or am I misinterpreting?